### PR TITLE
TIDY: Implementing empty title tag linter.

### LINKED ISSUE #16323: Report empty title tags as a style violation
It's easy to create new tests with the default `<title></title>` and forget to give the test a meaningful title. We should make `./mach test-tidy` report this as an error by adding a new method to tidy.py like `check_toml` that iterates over the lines of any html file.

Code: `python/tidy/servo_tidy/tidy.py`
Tests: add a testcase to `python/tidy/servo_tidy_tests/`, then add a method to `test_tidy.py` that verifies that the error is reported. Run it with `./mach test-tidy --self-test`.

### PR DESCRIPTION
**Pull request by zehcnas34**
I've added a tidy check method that yields an error when it comes across an empty `title` tag in HTML files. This pull request contains the changes specified in pull request #16362.

- [x] Implement linter.
- [x] Create tests for linter.
- [x] Remove empty title tags from existing html files.
- [x] Squash commits and rebase to master.
---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #16323

- [x] There are tests for these changes 

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/16362)

### CODE REVIEW

File: tests/wpt/web-platform-tests/content-security-policy/style-src/style-src-3_1.html [hunk 1]
```diff
@@ -0,0 +1,33 @@
+<!doctype html>
```
Comment (jdm) [inline tests/wpt/web-platform-tests/content-security-policy/style-src/style-src-3_1.html]: Why did this file return from the grave?

File: tests/wpt/web-platform-tests/content-security-policy/style-src/style-src-3_4.html [hunk 2]
```diff
@@ -0,0 +1,27 @@
+<!doctype html>
```
Comment (jdm) [inline tests/wpt/web-platform-tests/content-security-policy/style-src/style-src-3_4.html]: Why did this file come back?

File: tests/wpt/web-platform-tests/cssom/CSSStyleRule.html [hunk 3]
```diff
@@ -1,90 +1,24 @@
-<!DOCTYPE html>
```
Comment (jdm) [inline tests/wpt/web-platform-tests/cssom/CSSStyleRule.html]: The changes to this file should be reverted.

File: etc/ci/performance/page_load_test/example/example_async.html [hunk 4]
```diff
@@ -3,7 +3,7 @@
   <head>
     <meta charset="utf-8" />
     <meta name="viewport" content="width=device-width" />
-    <title></title>
+
```
Comment (jdm) [inline etc/ci/performance/page_load_test/example/example_async.html]: Rather than replacing with a blank line, it would be better to remove the line entirely (and in all other modified files).

File: tests/wpt/web-platform-tests/webgl/conformance-1.0.3/conformance/glsl/misc/expression-list-in-declarator-initializer.html [hunk 5]
```diff
@@ -28,12 +28,12 @@
...
 <link rel="stylesheet" href="../../resources/glsl-feature-tests.css" />
-<script src=/resources/testharness.js></script>
-<script src=/resources/testharnessreport.js></script>
+<script src=/resources/testharness.js></script>
+<script src=/resources/testharnessreport.js></script>
 <script src="../../../resources/js-test-pre.js"></script>
 <script src="../../resources/webgl-test-utils.js"></script>
 <script src="../../resources/glsl-conformance-test.js"></script>
-<title></title>
+
```
Comment (jdm) [inline tests/wpt/web-platform-tests/webgl/conformance-1.0.3/conformance/glsl/misc/expression-list-in-declarator-initializer.html]: The changes to the webgl/ directory are a problem, because these are tests that are automatically imported from an upstream repository. Can we tell the linter to ignore files in this directory instead, and revert the changes to them?

**Review by jdm — CHANGES_REQUESTED** [review]
I haven't looked at the actual lint; only the files that needed modification to pass it.

**Comment by Wafflespeanut** [comment]
Is this still being worked on?

File: tests/wpt/web-platform-tests/conformance-checkers/html-svg/0001isvalid.html [hunk 6]
```diff
@@ -14,7 +14,7 @@
 
     <!-- null-content elements -->
     <svg>
-
+      <title></title>
```
Comment (jdm) [inline tests/wpt/web-platform-tests/conformance-checkers/html-svg/0001isvalid.html]: Doesn't this complain now?

File: tests/wpt/web-platform-tests/conformance-checkers/html/elements/title/empty-novalid.html [hunk 7]
```diff
@@ -1,4 +1,3 @@
 <!doctype html>
 <!-- document with empty title element -->
 <meta charset=utf-8>
-<title></title>
```
Comment (jdm) [inline tests/wpt/web-platform-tests/conformance-checkers/html/elements/title/empty-novalid.html]: This test is explicitly testing empty title elements. Looks like we need to do something smarter.

File: tests/wpt/web-platform-tests/cssom/CSSStyleRule.html [hunk 8]
```diff
@@ -1,4 +1,4 @@
...
+<!doctype html>
```
Comment (jdm) [inline tests/wpt/web-platform-tests/cssom/CSSStyleRule.html]: Lowercase is treated differently than uppercase DOCTYPE.

**Review by jdm — CHANGES_REQUESTED** [review]
I have realized a problem - there is nothing preventing upstream tests from being added with empty title tags, which will cause trouble when we import them from upstream.

**Comment by jdm** [comment]
Given the problem I mention, I can't think of a way to enforce this check in a meaningful way, since I care most about the tests that we create and add to the tests/wpt/web-platform-tests/ directory. Sorry about putting you through this @ZehCnaS34 :(
